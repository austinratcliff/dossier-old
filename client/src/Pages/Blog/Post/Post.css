.Post {
  font-family: 'IBM Plex Mono', monospace;
  margin-top: 25px;
}

.Post:last-child {
  margin-bottom: 25px;
}

.Post .title {
  font-size: calc(10px + 2vmin);
  margin: 0;
}

.Post .date {
  font-size: calc(10px + 1vmin);
  margin: 0;
}

.Post .highlight-hover:hover {
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 50%, #92A8D1 50%);
  cursor: pointer;
}

.modal-backdrop {
  background-color: #111 !important;
}

.modal-content {
  border: 3px solid #111 !important;
  border-radius: 0 !important;
  background-color: white !important;
  color: #111 !important;
}

.modal-header {
  border-bottom: 3px solid #111 !important;
  font-family: 'Ibm Plex Mono', monospace !important;
}

.modal-body {
  font-family: 'IBM Plex Sans', sans-serif !important;
}

.modal-body h1,
.modal-body h2,
.modal-body h3,
.modal-body h4,
.modal-body h5,
.modal-body h6 {
  font-family: 'Ibm Plex Mono', monospace !important;
}

.modal-body a {
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 50%, #92A8D1 50%) !important;
  color: #111 !important;
}

.modal-body a:hover {
  color: #111 !important;
  text-decoration: none !important;
}

.modal-body img {
  display: block !important;
  margin: 0 auto !important;
}

.modal-body pre {
  border: 1px solid #111 !important;
  background-color: #111 !important;
  color: white !important;
  font-weight: bold !important;
  letter-spacing: 1.5px !important;
  display: flex !important;
  justify-content: center !important;
  align-self: center !important;
  width: 100% !important;
  padding: 15px 0 !important;
}

.modal-body code {
  font-family: 'Ibm Plex Mono', monospace !important;
}

.modal-body::-webkit-scrollbar {
  display: none !important;
}
