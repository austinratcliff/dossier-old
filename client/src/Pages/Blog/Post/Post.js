import React, { useState } from 'react'
import Modal from 'react-bootstrap/Modal'
import ReactMarkdown from 'react-markdown'
import './Post.css'

const Post = props => {
  const [show, setShow] = useState({ [props.post.id]: false })

  const handleShow = () => setShow({ ...show, [props.post.id]: true })
  const handleHide = () => setShow({ ...show, [props.post.id]: false })

  const date = new Date(props.post.published_at)
  const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
  const month = months[date.getMonth()]
  const day = date.getDate() < 10 ? '0' + date.getDate() : date.getDate()
  const year = date.getFullYear()

  return (
    <div className="Post">
      <h2
        className="title"
        onClick={handleShow}
      >
        <span className="highlight-hover">
          {props.post.title}
        </span>
      </h2>
      <p className="date">
        {month + ' ' + day + ', ' + year}
      </p>
      <Modal
        show={show[props.post.id]}
        onHide={handleHide}
        animation={false}
        size="lg"
        centered
        scrollable
      >
        <Modal.Header>
          <Modal.Title>
            {props.post.title}
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <ReactMarkdown source={props.post.body_markdown} />
        </Modal.Body>
      </Modal>
    </div>
  )
}

export default Post
