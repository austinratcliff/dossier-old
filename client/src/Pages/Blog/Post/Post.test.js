import React from 'react'
import ReactDOM from 'react-dom'
import ReactTestRenderer from 'react-test-renderer'
import Post from './Post'

describe('Post component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Post />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('matches snapshot', () => {
    const component = ReactTestRenderer.create(<Post />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
