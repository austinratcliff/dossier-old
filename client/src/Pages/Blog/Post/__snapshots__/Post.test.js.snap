// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component matches snapshot 1`] = `
<div
  className="Post"
>
  <h2
    className="title"
    onClick={[Function]}
  >
    <span
      className="highlight-hover"
    />
  </h2>
  <p
    className="date"
  >
    undefined NaN, NaN
  </p>
</div>
`;
